js/views/PageNav.js

```diff
@@ -121,24 +142,61 @@ export default class extends View {
     }
   }
 
+  onMouseEnterConnectedServerListItem() {
+    this.overConnectedServerListItem = true;
+    this.$connManagementContainer.addClass('open');
+  }
+
+  onMouseLeaveConnectedServerListItem() {
+    this.overConnectedServerListItem = false;
+
+    setTimeout(() => {
+      if (!this.overConnManagementContainer) {
+        this.$connManagementContainer.removeClass('open');
+      }
+    }, 100);
+  }
+
+  onMouseEnterConnManagementContainer() {
+    this.overConnManagementContainer = true;
+  }
+
+  onMouseLeaveConnManagementContainer() {
+    this.overConnManagementContainer = false;
+
+    setTimeout(() => {
+      if (!this.overConnectedServerListItem) {
+        this.$connManagementContainer.removeClass('open');
+      }
+    }, 100);
+  }
+
+  onNavListItemClick() {
+    this.closePopMenu();
+  }
+
   navListBtnClick() {
-    const $popMenu = this.$navList.hasClass('open') ? '' : this.$navList;
-    this.togglePopMenu($popMenu);
+    this.togglePopMenu();
   }
 
-  togglePopMenu($popMenu) {
-    if ($popMenu) {
-      this.$popMenus.not($popMenu).removeClass('open');
-      $popMenu.toggleClass('open');
-      this.$navOverlay.addClass('open');
-    } else {
-      this.$popMenus.removeClass('open');
-      this.$navOverlay.removeClass('open');
+  togglePopMenu() {
+    this.$navList.toggleClass('open');
+    this.$navOverlay.toggleClass('open');
+
+    if (!this.$navList.hasClass('open')) {
+      this.$connManagementContainer.addClass('open');
```

Comment:
Isn't this adding the open class to the connection management container if the menu is closed?

It can result in the connection management container being open by itself when the menu is gone if you open it, and click away on the page. 

![image](https://cloud.githubusercontent.com/assets/1584275/21992421/621e1558-dbe4-11e6-9cbf-fbe86cd0c8fd.png)

Can it just be removed?